extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -14,29 +17,61 @@ import {
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Uses the specified validation configuration to run validation against
+ * the user's shopping configuration.
+ * @param {!ShoppingConfigDataDef} productConfig The user's config object.
+ * @return {boolean} returns a boolean indicating whether the validation
+ *     was successful.
+ */
+function validateConfig(productConfig) {
+  let isValidConfig = true;
+  const errors = validateProduct(productConfig);
+
+  if (errors.length > 0) {
+    isValidConfig = false;
+    user().warn('AMP-STORY-SHOPPING-CONFIG', `${errors}`);
+  }
+
+  return isValidConfig;
+}
+
 /** @typedef {!Object<string, !ShoppingConfigDataDef> */
-export let KeyedShoppingConfigDef;
+let KeyedShoppingConfigDef;
 
 /**
- * Gets Shopping config from an element.
- * The config is validated and keyed by 'product-tag-id'.
- * @param {!Element} element <amp-story-shopping-attachment>
- * @return {!Promise<!KeyedShoppingConfigDef>}
+ * Validates and returns the shopping config corresponding to the given
+ * amp-story-shopping-attachment element.
+ * @param {!Element} shoppingAttachmentEl <amp-story-shopping-attachment>
+ *     The amp story shopping attachment element
+ * @return {!Promise<!KeyedShoppingConfigDef>} An object with product ID
+ *     keys that each have a `ShoppingConfigDataDef` value
  */
-export function getShoppingConfig(element) {
-  return getElementConfig(element).then((config) => {
-    //TODO(#36412): Add call to validate config here.
-    return keyByProductTagId(config);
+export function getShoppingConfig(shoppingAttachmentEl) {
+  return getElementConfig(shoppingAttachmentEl).then((config) => {
+    const allItems = config['items'];
+    const validItems = allItems.filter((item) => validateConfig(item));
+    if (allItems.length != validItems.length) {
+      user().warn(
+        'AMP-STORY-SHOPPING-CONFIG',
+        `Please fix: ${
+          allItems.length - validItems.length
+        } product(s) have invalid shopping configuration values. See the error messages above for more details.`
+      );
+    }
```

Comment:
Nit: This warning is redundant in my opinion. I would just remove it.
```suggestion
    if (allItems.length != validItems.length) {
      user().warn(
        'AMP-STORY-SHOPPING-CONFIG',
        `Please fix: ${
          allItems.length - validItems.length
        } product(s) have invalid shopping configuration values. See the error messages above for more details.`
      );
    }
```